sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubClientBuilderTest.java

```diff
@@ -125,6 +127,27 @@ public void testProxyOptionsConfiguration(String proxyConfiguration, boolean exp
         Assertions.assertEquals(expectedClientCreation, clientCreated);
     }
 
+    @Test
```

Comment:
For this one, I would add an IntegrationTest that takes an AzureNameKeyCredential since we can't really mock the entire flow.

See: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/IntegrationTestBase.java#L222

Where we get ConnectionStringProperties, you could use that to create your own AzureNameKeyCredential and test that we can send it.

Similarly with the SAS Credential.

Reply:
According to your comment in the new revision amendments have been made.